lib/geo_combine/ogp.rb

```diff
@@ -0,0 +1,131 @@
+module GeoCombine
+  # Data model for ESRI's open data portal metadata
+  class OGP
+    include GeoCombine::Formatting
+    attr_reader :metadata
+
+    ##
+    # Initializes an OGP object for parsing
+    # @param [String] metadata a valid serialized JSON string from an ESRI Open
+    # Data portal
+    def initialize(metadata)
+      @metadata = JSON.parse(metadata)
+    end
+
+    ##
+    # Creates and returns a Geoblacklight schema object from this metadata
+    # @return [GeoCombine::Geoblacklight]
+    def to_geoblacklight
+      GeoCombine::Geoblacklight.new(geoblacklight_terms.to_json)
+    end
+
+    ##
+    # Builds a Geoblacklight Schema type hash from Esri Open Data portal
+    # metadata
+    # @return [Hash]
+    def geoblacklight_terms
+      {
+        # Required fields
+        dc_identifier_s: metadata['LayerId'],
+        layer_slug_s: sluggify(metadata['LayerId']),
+        dc_title_s: metadata['LayerDisplayName'],
+        solr_geom: envelope,
+        dct_provenance_s: metadata['Institution'],
+        dc_rights_s: metadata['Access'],
+        geoblacklight_version: '1.0',
+
+        # Recommended fields
+        dc_description_s: metadata['Abstract'],
+        layer_geom_type_s: ogp_geom,
+        dct_references_s: references,
+        layer_id_s: "#{metadata['Workspace']}#{metadata['Name']}",
+
+        # Optional
+        dct_temporal_sm: [metadata['ContentDate']],
+        dc_format_s: ogp_formats,
+        # dct_issued_dt
+        # dc_language_s
+        dct_spatial_sm: metadata['PlaceKeywords'].split(' '),
+        solr_year_i: Time.parse(metadata['ContentDate']).year,
+        dc_publisher_s: metadata['Publisher'],
+        dc_subject_sm: metadata['ThemeKeywords'].split(' '),
```

Comment:
The `PlaceKeywords` and `ThemeKeywords` are likely not extractable from the OGP schema because they are already tokenized. The best way to extract them is to get them from the FGDC metadata in the `FgdcText` field.